File: src/java/gasolina.java
// Calcular el precio de la gasolina - Java

package java;

import java.util.Scanner;

public class gasolina {

    /**
     * Calcula los litros que se gastan en un viaje
     *
     * @param km       Kilómetros del viaje
     * @param litroskm Litros que gasta el coche por kilómetro
     * @return Litros que gastará en el viaje
     */
    static float calcularLitros(float km, float litroskm) {
        return km * litroskm;
    }

    /**
     * Calcula el precio de la gasolina
     *
     * @param litros Litros que se gastan
     * @param precio Precio de la gasolina
     * @return Precio del viaje
     */
    static float calcularPrecio(float litros, float precio) {
        return litros * precio;
    }

    /**
     * Calcula el precio por persona
     *
     * @param precio   Precio del viaje
     * @param personas Número de personas
     * @return Precio por persona
     */
    static float precioPersona(float precio, int personas) {
        return precio / personas;
    }

    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
            float precio, km, litroskm;
            int personas;

            System.out.println("Introduce los kilómetros del viaje: ");
            km = scanner.nextFloat();
            System.out.println("Introduce los litros que gasta el coche por kilómetro: ");
            litroskm = scanner.nextFloat();
            System.out.println("Introduce el precio de la gasolina: ");
            precio = scanner.nextFloat();
            System.out.println("Introduce el número de personas: ");
            personas = scanner.nextInt();

            float litros = calcularLitros(km, litroskm);
            float precioTotal = calcularPrecio(litros, precio);
            float precioPersona = precioPersona(precioTotal, personas);

            System.out.println("El precio total del viaje es de " + precioTotal + "€");
            System.out.println("El precio por persona es de " + precioPersona + "€");
        }
    }

}

File: src/c++/gasolina.cpp
/**
 * @file gasolina.cpp
 * @author Carlota de la Vega
 * @brief Programa que calcula el precio de la gasolina por persona.
 * @version 0.1
 * @date 2023-08-28
 *
 * @copyright Copyright (c) 2023
 *
 */

#include <iostream>
#include <string>

using namespace std;

/**
 * @brief Función que calcula los litros que se gastan en un viaje.
 *
 * @param km Kilómetros totales del trayecto.
 * @param litroskm Litros que gasta el coche por kilómetro.
 * @return float litros totales.
 */
float calcularLitros(float km, float litroskm) {
    return km * litroskm;
}

/**
 * @brief Función que calcula el precio total de la gasolina.
 *
 * @param litros Litros gastados en el viaje.
 * @param precio Precio de la gasolina.
 * @return float precio total.
 */
float calcularPrecio(float litros, float precio) {
    return litros * precio;
}

/**
 * @brief Función que calcula el precio por persona.
 *
 * @param precio Precio total de la gasolina.
 * @param personas Personas que viajan.
 * @return float precio por persona.
 */
float precioPersona (float precio, int personas) {
    return precio / personas;
}

int main() {
    float precio, litros, personas, km;

    cout << "Ingrese la distancia total del viaje en km: ";
    cin >> km;
    cout << "Ingrese el precio de la gasolina: ";
    cin >> precio;
    cout << "Ingrese la cantidad de litros que gasta por kilómetro: ";
    cin >> litros;
    cout << "Infrese la cantidad de personas que viajan: ";
    cin >> personas;

    float litrosTotales = calcularLitros(km, litros);
    float precioTotal = calcularPrecio(litrosTotales, precio);
    float precioPersonaTotal = precioPersona(precioTotal, personas);

    cout << "El total a pagar por persona es: " << precioPersonaTotal << endl;

    return 0;
}